### PR TITLE
RFC: vars/buildImage: staging -> production

### PR DESCRIPTION
**Pull request by khilman**
@gctucker You've already done some work in unifying tokens between staging and master.  Is there  a better way to do this so we have a job that works on both staging and production?

### CODE REVIEW

File: vars/buildImage.groovy [hunk 1]
```diff
@@ -92,9 +92,9 @@ def makeImageStep(String pipeline_version, String arch, String debian_arch, Stri
                 }
 
                 stage("Upload images for ${arch}") {
-                    withCredentials([string(credentialsId: 'Staging KernelCI API Token', variable: 'API_TOKEN')]) {
```
Comment (gctucker) [inline vars/buildImage.groovy]: With the build and bisection jobs, I've set up a parameter so the token name can be stored in the job configuration:
https://github.com/kernelci/lava-ci/blob/master/jenkins/bisect.jpl#L203
```
        withCredentials([string(credentialsId: params.KCI_TOKEN_ID,
                                variable: 'SECRET')]) {
```
Same thing in the build job:
https://github.com/kernelci/kernelci-build/blob/master/jenkins/build.jpl#L98

File: vars/buildImage.groovy [hunk 2]
```diff
@@ -92,9 +92,9 @@ def makeImageStep(String pipeline_version, String arch, String debian_arch, Stri
...
+                    withCredentials([string(credentialsId: 'KernelCI Upload API Token', variable: 'API_TOKEN')]) {
                         sh """
-                            python push-source.py --token ${API_TOKEN} --api https://staging-api.kernelci.org \
```
Comment (gctucker) [inline vars/buildImage.groovy]: Same thing for the API URL, there's a job parameter for that in build and bisection jobs:

https://github.com/kernelci/kernelci-build/blob/master/jenkins/build.jpl#L101
```
API=${params.KCI_API_URL} \
TOKEN=${SECRET} \
${kci_build}/build.py -i ${opt} -c ${defconfig}
```

**Comment by gctucker** [comment]
Hi @khilman, @ana, I've put some review comments on the changes.  Essentially I've been using standard Jenkins job parameters for that, so the jobs can be set up on any Jenkins server with different default values.  It also allows manual runs with alternative values, say if you want to try something with a different backend.

The downside is that the number of parameters can grow quite quickly, so it can be a bit harder to manage than a config file, but it's working well enough for now.  In the future, we may just have one parameter with a URL where the job would fetch a config file from (or git clone a config repo).

**Comment by ana** [comment]
I'd love to avoid using jenkins job parameters but I don't see a better option for now.
In the near future, the idea of using a single job parameter with an URL is very worth pursuing, we should look at this when we eventually consolidate all the jenkins pipelines code.

**Comment by khilman** [comment]
@ana can you follow up with this and propose some changes using jenkins job parameters that will work in both production and staging?

**Comment by ana** [comment]
Yes, I'll take care of this.

**Comment by ana** [comment]
@khilman you can close this now.

**Comment by khilman** [comment]
Was this fixed somewhere else?

**Comment by ana** [comment]
Yes, with https://github.com/kernelci/kernelci-build-staging/pull/36

**Comment by khilman** [comment]
ok

**Pull request closed without merging**